pandas/tests/indexing/multiindex/test_multiindex.py

```diff
@@ -98,3 +98,25 @@ def test_multiindex_with_datatime_level_preserves_freq(self):
         result = df.loc[0].index
         tm.assert_index_equal(result, dti)
         assert result.freq == dti.freq
+
+    def test_multiindex_complex(self):
+        # GH#42145
+        complex_data = [1 + 2j, 4 - 3j, 10 - 1j, 4 + 5j, 5 - 8j]
+        non_complex_data1 = [3, 4, 5, 6, 7]
+        non_complex_data2 = [0.1, 0.2, 0.3, 0.4, 0.5]
+        result = DataFrame(
+            {
+                "x": complex_data,
+                "y": non_complex_data1,
+                "z": non_complex_data2,
```

Comment:
Could you simply use 1 here, makes it easier to understand since this is not bug relevant, isn‘t it?

Reply:
amended